modules/transactionpool/transactionpool_test.go

```diff
@@ -336,3 +336,327 @@ func TestFeeEstimation(t *testing.T) {
+	for i := 0; i < numGraphs; i++ {
+		var edges []types.TransactionGraphEdge
+
+		// Create the root of the graph.
+		feeValues := types.SiacoinPrecision
+		firstRowValues := graphFund.Sub(feeValues.Mul64(uint64(rows))).Div64(uint64(rows))
+		for j := 0; j < rows; j++ {
+			edges = append(edges, types.TransactionGraphEdge{
+				Dest:   j + 1,
+				Fee:    types.SiacoinPrecision,
+				Source: 0,
+				Value:  firstRowValues,
+			})
+		}
+
+		// Create each row of the graph.
+		var firstNodeValue types.Currency
+		nodeIndex := 1
+		for j := 0; j < rows; j++ {
+			// Create the first node in the row, which has an increasing
+			// balance.
+			rowValue := firstRowValues.Sub(types.SiacoinPrecision.Mul64(uint64(j + 1)))
+			firstNodeValue = firstNodeValue.Add(rowValue)
+			edges = append(edges, types.TransactionGraphEdge{
+				Dest:   nodeIndex + (rows - j),
+				Fee:    types.SiacoinPrecision,
+				Source: nodeIndex,
+				Value:  firstNodeValue,
+			})
+			nodeIndex++
+
+			// Create the remaining nodes in this row.
+			for k := j + 1; k < rows; k++ {
+				edges = append(edges, types.TransactionGraphEdge{
+					Dest:   nodeIndex + (rows - (j + 1)),
+					Fee:    types.SiacoinPrecision,
+					Source: nodeIndex,
+					Value:  rowValue,
+				})
+				nodeIndex++
+			}
+		}
+
+		// Build the graph and add it to the stack of graphs.
+		graph, err := types.TransactionGraph(finalTxn.SiacoinOutputID(uint64(i)), edges)
+		if err != nil {
+			t.Fatal(err)
+		}
+		graphs = append(graphs, graph)
+	}
+
+	// Add all of the root transactions to the blockchain to throw off the
+	// parent math off for the transaction pool.
+	for _, graph := range graphs {
+		err := tpt.tpool.AcceptTransactionSet([]types.Transaction{graph[0]})
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	_, err = tpt.miner.AddBlock()
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Add all of the transactions in each graph into the tpool, one transaction
+	// at a time, interweaved, chaotically.
+	for i := 1; i < len(graphs[0]); i++ {
+		for j := 0; j < len(graphs); j++ {
+			err := tpt.tpool.AcceptTransactionSet([]types.Transaction{graphs[j][i]})
+			if err != nil {
+				t.Fatal(err, i, j)
+			}
+		}
+	}
+
+	// Mine blocks until the tpool is gone.
+	for tpt.tpool.transactionListSize > 0 {
+		_, err := tpt.miner.AddBlock()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+}
+
+func TestHeapFees(t *testing.T) {
+	if testing.Short() {
+		t.SkipNow()
+	}
+	tpt, err := createTpoolTester(t.Name())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tpt.Close()
+
+	// Mine a few more blocks to get some extra funding.
+	for i := 0; i < 5; i++ {
+		_, err := tpt.miner.AddBlock()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
+	// Make a bunch of transactions with high fees. A bunch with low fees.
+	// See that the next block has all high fee txns.
+	// Iterate over block txns and add up fees
+	// Prepare a bunch of outputs for a series of graphs to fill up the
+	// transaction pool.
+	coinFrac := types.SiacoinPrecision
```

Comment:
Comments should apply to code blocks, this comment + code block is completely disconnected from whatever is actually happening.